app/scripts/migrations/117.ts

```diff
@@ -1,62 +0,0 @@
-import { cloneDeep } from 'lodash';
-import {
-  TransactionMeta,
-  TransactionStatus,
-  TransactionError,
-} from '@metamask/transaction-controller';
-
-type VersionedData = {
-  meta: { version: number };
-  data: Record<string, unknown>;
-};
-
-export const version = 117;
-
-// Target date is December 8, 2023 - 00:00:00 UTC
-export const TARGET_DATE = new Date('2023-12-08T00:00:00Z').getTime();
-
-const STUCK_STATES = [TransactionStatus.approved, TransactionStatus.signed];
-
-type FailedTransactionMeta = TransactionMeta & {
-  status: TransactionStatus.failed;
-  error: TransactionError;
-};
-
-export const StuckTransactionError = {
-  name: 'StuckTransactionDueToStatus',
-  message: 'Transaction is stuck due to status - migration 116',
-};
-
-/**
- * This migration sets the `status` to `failed` for all transactions created before December 8, 2023 that are still `approved` or `signed`.
- *
- * @param originalVersionedData
- */
-export async function migrate(
-  originalVersionedData: VersionedData,
-): Promise<VersionedData> {
-  const versionedData = cloneDeep(originalVersionedData);
-  versionedData.meta.version = version;
-  transformState(versionedData.data);
-  return versionedData;
-}
-
-// TODO: Replace `any` with type
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function transformState(state: Record<string, any>) {
-  const transactions: TransactionMeta[] =
-    state?.TransactionController?.transactions ?? [];
-
-  for (const transaction of transactions) {
-    if (
-      transaction.time < TARGET_DATE &&
-      STUCK_STATES.includes(transaction.status)
-    ) {
-      transaction.status = TransactionStatus.failed;
-
-      const failedTransaction = transaction as FailedTransactionMeta;
-
-      failedTransaction.error = StuckTransactionError;
-    }
-  }
-}
```

Comment:
Shouldn't the revert PR only revert the 115.ts file which was responsible to migrate users ti linea sepolia?
I think 117.ts and 116.ts logic should still stay and maybe be renamed if we are removing 115.ts?

Reply:
In this PR, we address changes to our migration files. Specifically, we've removed the migration file named `115.ts`. During the period since its creation, two additional migration files, `116.ts` and `117.ts`, were introduced. To maintain a sequential order and fill the gap left by the deletion of `115.ts`, we've renumbered these files: `116.ts` has been renamed to `115.ts`, and `117.ts` has been renamed to `116.ts`. This adjustment ensures our migration files are sequentially organized without any missing numbers in the sequence.

Reply:
Renaming migration files does not seem a good idea as it might lead to migrations being skipped for anyone that is between 115 and 117 (inclusive). For example this should prevent anyone already on version 117 to skip the future 117 migration file when it is added again.

Reply:
hey @cryptotavares ,
I agree with your points, but it's important to note a crucial detail: the migration file in question has never been deployed to a production environment, nor has it been run at all.

Reply:
@salimtb it has been run in develop, meaning that everyone's local env might be affected. Example: if I was to add an 117 migration right now, I would not be able to trigger it locally without modifying the extension persisted state.

(output from `window.stateHooks.getCleanAppState().then(console.log)` in the browser console)
![Screenshot 2024-05-07 at 10 21 51](https://github.com/MetaMask/metamask-extension/assets/15957235/5918c169-2d29-4b36-bfb0-17d77925a11a)

It has also been included in the 11.16 release cut. We can cherry pick a fix into 11.16 release candidate.. but I suspect that anyone that is already QAing that branch might face similar issues, if we end up having to cherry pick a commit to add another migration (future 117).

In my opinion, it would be cleaner and simpler to just add another migration (lets say 118) that would revert 115. But I don't know how we have approach this in the past. @Gudahtt or @danjm, how have we dealt with this in the past?

Reply:
eeemm ok , thnx for the clarification @cryptotavares , i'll add another file ( 118.ts ) like you just suggested.
will ping you in a few moment once it's done.
thnx for your comment and clarification

Reply:
@cryptotavares update done , pls check again

Reply:
@cryptotavares I agree that it is preferable to avoid removing already committed migration files when we can. At the same time, I don't like having migration code that does nothing.

We faced a similar issue about a year ago, and because the changes were not yet on master, we decided to renumber the migration files https://consensys.slack.com/archives/CTQAGKY5V/p1677784034578949

Reply:
I didn't see this thread before, but Salim flagged it for me after I opened this https://github.com/MetaMask/metamask-extension/pull/24666